### ISSUE TITLE
Duplicate Point IDs in point cloud data

### ISSUE THREAD

**Issue by juj** [google-ar/arcore-android-sdk#630]
It looks like point cloud data contains duplicate entries of the same ID. Consider the following patch to the `hello_ar_c` sample:

```diff
diff --git a/samples/hello_ar_c/app/src/main/cpp/hello_ar_application.cc b/samples/hello_ar_c/app/src/main/cpp/hello_ar_application.cc
index 84e5142..86b5c61 100644
--- a/samples/hello_ar_c/app/src/main/cpp/hello_ar_application.cc
+++ b/samples/hello_ar_c/app/src/main/cpp/hello_ar_application.cc
@@ -269,7 +269,29 @@ void HelloArApplication::OnDrawFrame() {
   ArStatus point_cloud_status =
       ArFrame_acquirePointCloud(ar_session_, ar_frame_, &ar_point_cloud);
   if (point_cloud_status == AR_SUCCESS) {
+      int32_t numPoints = 0;
+      ArPointCloud_getNumberOfPoints(ar_session_, ar_point_cloud, &numPoints);
+      if (numPoints > 0) {
+          const int32_t *points = 0;
+          ArPointCloud_getPointIds(ar_session_, ar_point_cloud, &points);
+          const float *data = 0;
+          ArPointCloud_getData(ar_session_, ar_point_cloud, &data);
+
+          if (points && data)
+          {
+              for(int i = 0; i < numPoints; ++i)
+                  for(int j = i+1; j < numPoints; ++j)
+                      if (points[i] == points[j])
+                          LOGE("ERROR: Point cloud has duplicate ID value %d at indices %d and %d. First point: %f,%f,%f,%f. Second point: %f,%f,%f,%f",
+                               points[i], i, j,
+                               data[4*i], data[4*i+1], data[4*i+2], data[4*i+3],
+                               data[4*j], data[4*j+1], data[4*j+2], data[4*j+3]
+                          );
+          }
+      }

     point_cloud_renderer_.Draw(projection_mat * view_mat, ar_session_,
                                ar_point_cloud);
     ArPointCloud_release(ar_point_cloud);
   }
```

Running the `hello_ar_c` application with this change will start spamming the following to console

```
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 22205 at indices 46 and 47. First point: 0.183718,-0.837910,-1.518995,0.253496. Second point: 0.179898,-0.841578,-1.524095,0.412219
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 26023 at indices 86 and 87. First point: 0.374022,-0.888579,-1.822928,0.365505. Second point: 0.380711,-0.910654,-1.866604,0.401044
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 38878 at indices 178 and 179. First point: 0.782685,-0.830376,-1.121392,0.256787. Second point: 0.785613,-0.836513,-1.131951,0.298210
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 40224 at indices 188 and 189. First point: 0.622305,-0.807652,-1.078088,0.000000. Second point: 0.636300,-0.836004,-1.119867,0.281498
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 41959 at indices 232 and 233. First point: 0.342934,-0.965950,-1.042583,0.280620. Second point: 0.333880,-0.902544,-0.967333,0.000000
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 42012 at indices 236 and 237. First point: 0.765187,-0.749619,-1.976374,0.084852. Second point: 0.777226,-0.766899,-2.024393,0.282953
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 42132 at indices 241 and 242. First point: 0.639233,-0.704650,-1.268963,0.000000. Second point: 0.652035,-0.726507,-1.313529,0.310960
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 42224 at indices 255 and 256. First point: 0.207559,-0.798891,-1.258052,0.000000. Second point: 0.207435,-0.872412,-1.379078,0.293784
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 42225 at indices 257 and 258. First point: 0.285283,-0.790416,-1.447516,0.000000. Second point: 0.281382,-0.768561,-1.407558,0.294079
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 42270 at indices 260 and 261. First point: 0.807816,-0.820683,-1.077744,0.000000. Second point: 0.851153,-0.875289,-1.157270,0.311852
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 22205 at indices 46 and 47. First point: 0.183718,-0.837910,-1.518995,0.253496. Second point: 0.179898,-0.841578,-1.524095,0.412219
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 26023 at indices 86 and 87. First point: 0.374022,-0.888579,-1.822928,0.365505. Second point: 0.380711,-0.910654,-1.866604,0.401044
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 38878 at indices 178 and 179. First point: 0.782685,-0.830376,-1.121392,0.256787. Second point: 0.785613,-0.836513,-1.131951,0.298210
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 40224 at indices 188 and 189. First point: 0.622305,-0.807652,-1.078088,0.000000. Second point: 0.636300,-0.836004,-1.119867,0.281498
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 41959 at indices 232 and 233. First point: 0.342934,-0.965950,-1.042583,0.280620. Second point: 0.333880,-0.902544,-0.967333,0.000000
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 42012 at indices 236 and 237. First point: 0.765187,-0.749619,-1.976374,0.084852. Second point: 0.777226,-0.766899,-2.024393,0.282953
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 42132 at indices 241 and 242. First point: 0.639233,-0.704650,-1.268963,0.000000. Second point: 0.652035,-0.726507,-1.313529,0.310960
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 42224 at indices 255 and 256. First point: 0.207559,-0.798891,-1.258052,0.000000. Second point: 0.207435,-0.872412,-1.379078,0.293784
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 42225 at indices 257 and 258. First point: 0.285283,-0.790416,-1.447516,0.000000. Second point: 0.281382,-0.768561,-1.407558,0.294079
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 42270 at indices 260 and 261. First point: 0.807816,-0.820683,-1.077744,0.000000. Second point: 0.851153,-0.875289,-1.157270,0.311852
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 22205 at indices 46 and 47. First point: 0.183718,-0.837910,-1.518995,0.253496. Second point: 0.179898,-0.841578,-1.524095,0.412219
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 26023 at indices 86 and 87. First point: 0.374022,-0.888579,-1.822928,0.365505. Second point: 0.380711,-0.910654,-1.866604,0.401044
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 38878 at indices 178 and 179. First point: 0.782685,-0.830376,-1.121392,0.256787. Second point: 0.785613,-0.836513,-1.131951,0.298210
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 40224 at indices 188 and 189. First point: 0.622305,-0.807652,-1.078088,0.000000. Second point: 0.636300,-0.836004,-1.119867,0.281498
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 41959 at indices 232 and 233. First point: 0.342934,-0.965950,-1.042583,0.280620. Second point: 0.333880,-0.902544,-0.967333,0.000000
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 42012 at indices 236 and 237. First point: 0.765187,-0.749619,-1.976374,0.084852. Second point: 0.777226,-0.766899,-2.024393,0.282953
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 42132 at indices 241 and 242. First point: 0.639233,-0.704650,-1.268963,0.000000. Second point: 0.652035,-0.726507,-1.313529,0.310960
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 42224 at indices 255 and 256. First point: 0.207559,-0.798891,-1.258052,0.000000. Second point: 0.207435,-0.872412,-1.379078,0.293784
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 42225 at indices 257 and 258. First point: 0.285283,-0.790416,-1.447516,0.000000. Second point: 0.281382,-0.768561,-1.407558,0.294079
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 42270 at indices 260 and 261. First point: 0.807816,-0.820683,-1.077744,0.000000. Second point: 0.851153,-0.875289,-1.157270,0.311852
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 22205 at indices 46 and 47. First point: 0.183718,-0.837910,-1.518995,0.253496. Second point: 0.179898,-0.841578,-1.524095,0.412219
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 26023 at indices 86 and 87. First point: 0.374022,-0.888579,-1.822928,0.365505. Second point: 0.380711,-0.910654,-1.866604,0.401044
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 38878 at indices 178 and 179. First point: 0.782685,-0.830376,-1.121392,0.256787. Second point: 0.785613,-0.836513,-1.131951,0.298210
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 40224 at indices 188 and 189. First point: 0.622305,-0.807652,-1.078088,0.000000. Second point: 0.636300,-0.836004,-1.119867,0.281498
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 41959 at indices 232 and 233. First point: 0.342934,-0.965950,-1.042583,0.280620. Second point: 0.333880,-0.902544,-0.967333,0.000000
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 42012 at indices 236 and 237. First point: 0.765187,-0.749619,-1.976374,0.084852. Second point: 0.777226,-0.766899,-2.024393,0.282953
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 42132 at indices 241 and 242. First point: 0.639233,-0.704650,-1.268963,0.000000. Second point: 0.652035,-0.726507,-1.313529,0.310960
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 42224 at indices 255 and 256. First point: 0.207559,-0.798891,-1.258052,0.000000. Second point: 0.207435,-0.872412,-1.379078,0.293784
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 42225 at indices 257 and 258. First point: 0.285283,-0.790416,-1.447516,0.000000. Second point: 0.281382,-0.768561,-1.407558,0.294079
E hello_ar_example_c: ERROR: Point cloud has duplicate ID value 42270 at indices 260 and 261. First point: 0.807816,-0.820683,-1.077744,0.000000. Second point: 0.851153,-0.875289,-1.157270,0.311852
```

This does not seem correct. Each point should have a unique ID? Note that this is not looking at two point clouds from different frames, but data returned within a single point cloud.

Tested on Huawei Nexus 6P (Android 8.1.0, API 27) with

```
I native  : version_check.cc:23 ARCore Version: APK version:1.5.180910096
I native  : version_check.cc:24 ARCore Version: emulated SDK version:1.5.180815000
I third_party/arcore/ar/core/android/sdk/session_create.cc: ARCore Version: SDK build name: 1.5
```

**Comment by inio** [comment]
Good catch!  Thanks for reporting this.

**Comment by inio** [comment]
This is fixed in ARCore 1.6 which was released yesterday.
